src/cpp/session/modules/SessionGit.cpp

```diff
@@ -204,10 +204,52 @@ std::string gitBin()
 }
 #endif
 
+bool waitForIndexLock(const FilePath& workingDir)
+{
+   using namespace boost::posix_time;
+   
+   // count the number of retries (avoid getting stuck in wait loops when
+   // an index.lock file exists and is never cleaned up)
+   static int retryCount = 0;
```

Comment:
That's indeed what I intended here -- maybe I can try to explain what I'm trying to accomplish. Suppose RStudio is working in a project where a stale `index.lock` has been left behind from an earlier (crashed?) Git process. In such a case, I didn't want to have the IDE always waiting for 1 second on future attempts to call into Git -- instead, it could just 'fail fast' without retrying anything.

This might be overly pessimistic as we don't call into git that frequently, and a one second pause would likely be imperceptible in the contexts where it is called -- let me know what you think.